src/content/en/updates/2017/12/chrome-loadtimes-deprecated.md

```diff
@@ -0,0 +1,358 @@
+  // If the browser supports the Navigation Timing 2 and HR Time APIs, use
+  // them, otherwise fall back to the Navigation Timing 1 API.
+  if (window.PerformanceNavigationTiming && performance.timeOrigin) {
+    const ntEntry = performance.getEntriesByType('navigation')[0];
+    return (ntEntry.startTime + performance.timeOrigin) / 1000;
+  } else {
+    return performance.timing.navigationStart / 1000;
+  }
+}
+```
+
+### `startLoadTime`
+
+```
+function startLoadTime() {
+  // If the browser supports the Navigation Timing 2 and HR Time APIs, use
+  // them, otherwise fall back to the Navigation Timing 1 API.
+  if (window.PerformanceNavigationTiming && performance.timeOrigin) {
+    const ntEntry = performance.getEntriesByType('navigation')[0];
+    return (ntEntry.startTime + performance.timeOrigin) / 1000;
+  } else {
+    return performance.timing.navigationStart / 1000;
+  }}
+```
+
+### `commitLoadTime`
+
+```
+function commitLoadTime() {
+  // If the browser supports the Navigation Timing 2 and HR Time APIs, use
+  // them, otherwise fall back to the Navigation Timing 1 API.
+  if (window.PerformanceNavigationTiming && performance.timeOrigin) {
+    const ntEntry = performance.getEntriesByType('navigation')[0];
+    return (ntEntry.responseStart + performance.timeOrigin) / 1000;
+  } else {
+    return performance.timing.responseStart / 1000;
+  }
+}
+```
+
+### `finishDocumentLoadTime`
+
+```
+function finishDocumentLoadTime() {
+  // If the browser supports the Navigation Timing 2 and HR Time APIs, use
+  // them, otherwise fall back to the Navigation Timing 1 API.
+  if (window.PerformanceNavigationTiming && performance.timeOrigin) {
+    const ntEntry = performance.getEntriesByType('navigation')[0];
+    return (ntEntry.domContentLoadedEventEnd + performance.timeOrigin) / 1000;
+  } else {
+    return performance.timing.domContentLoadedEventEnd / 1000;
+  }
+}
+```
+
+### `finishLoadTime`
+
+```
+function finishLoadTime() {
+  // If the browser supports the Navigation Timing 2 and HR Time APIs, use
+  // them, otherwise fall back to the Navigation Timing 1 API.
+  if (window.PerformanceNavigationTiming && performance.timeOrigin) {
+    const ntEntry = performance.getEntriesByType('navigation')[0];
+    return (ntEntry.loadEventEnd + performance.timeOrigin) / 1000;
+  } else {
+    return performance.timing.loadEventEnd / 1000;
+  }
+}
+```
+
+### `firstPaintTime`
+
+```
+function firstPaintTime() {
+  if (window.PerformancePaintTiming) {
+    const fpEntry = performance.getEntriesByType('paint')[0];
+    return (fpEntry.startTime + performance.timeOrigin) / 1000;
+  }
+}
+```
+
+### `firstPaintAfterLoadTime`
+
+```
+function firstPaintTimeAfterLoad() {
+  // This was never actually implemented and always returns 0.
+  return 0;
+}
+```
+
+### `navigationType`
+
+```
+function navigationType() {
+  if (window.PerformanceNavigationTiming) {
+    const ntEntry = performance.getEntriesByType('navigation')[0];
+    return ntEntry.type;
```

Comment:
These snippets are awesome. For ones like this that feature detect, they return `undefined` if you don't entire the conditional.

You could change them to always return something:

```js
function navigationType() {
  return performance.getEntriesByType('navigation')[0].type;
}
```

Reply:
Do you think returning `null` is better than `undefined`? In your example it would error if the browser didn't support `performance.getEntriesByType` or the "navigation" entry type.,

Reply:
I think everyone supports `performance.getEntriesByType` now. Oh, does `performance.getEntriesByType` throw if the type doesn't exist?

Reply:
Safari 10 (and I believe 10.1) do not. And it wouldn't throw if the entry type doesn't, but it would throw if you tried to access `.type` of `undefined`.

Reply:
Ack. If you want to include older browsers 👍